src/services/messagesService.ts

```diff
@@ -28,39 +31,47 @@ export default class MessagesService {
    * Retrieves all messages for a specific user.
    */
   public async getMessagesByUser(user: User): Promise<Messages> {
-    const messagesPayload = await this.apiClient
+    const response = await this.apiClient
       .getClient(user.fiscal_code)
-      .getMessagesByUser();
+      .getMessagesByUserWithHttpOperationResponse();
 
-    const errorOrApiMessages = GetMessagesByUserOKResponse.decode(
-      messagesPayload
-    );
-    if (isLeft(errorOrApiMessages)) {
-      const errorOrProblemJson = ProblemJson.decode(messagesPayload);
+    const simpleResponse = new SimpleHttpOperationResponse(response);
 
+    if (simpleResponse.status() !== 200) {
+      const errorOrProblemJson = ProblemJson.decode(
+        simpleResponse.bodyAsJson()
+      );
       if (isLeft(errorOrProblemJson)) {
-        winston.log(
-          "error",
-          "Unknown response from getMessagesByUser API:",
-          messagesPayload
+        winston.error(
+          "Unknown response from getMessagesByUser API: %s",
+          ReadableReporter.report(errorOrProblemJson)
         );
         throw new Error(messageErrorOnUnknownResponse);
-      } else {
-        winston.log(
-          "error",
-          "API error in getMessagesByUser:",
-          messagesPayload
-        );
+      }
+
+      if (simpleResponse.status() !== 404) {
```

Comment:
same here `simpleResponse.isStatusNotFound()` or even shorter since it's obvious we're talking about the status: `simpleResponse.isNotFound()`